### PR TITLE
Test covering

### PR DESCRIPTION
**Pull request by igorlima**
It's part of #3

### CODE REVIEW

File: generators/app/index.js [hunk 1]
```diff
@@ -1,114 +1,109 @@
...
-    this.log( this.destinationPath( this.classname + '.js') );
-    this.log( this.sourceRoot() );
-    this.log( this.templatePath('index.js') );
+    this.log(this.destinationRoot())
+    this.log(this.destinationPath(this.classname + '.js'))
+    this.log(this.sourceRoot())
+    this.log(this.templatePath('index.js'))
   },
 
   // The name `constructor` is important here
```
Comment (sashanka) [inline generators/app/index.js]: Why? Add more detail. Comments are not only for you to read the code better later but for the entire team. Please see JSDoc format to add comments.

Comment (sashanka) [inline generators/app/index.js]: https://google.github.io/styleguide/javascriptguide.xml?showone=Comments#Comments

File: generators/app/index.js [hunk 2]
```diff
@@ -1,114 +1,109 @@
-var generators = require('yeoman-generator');
-var _ = require('lodash');
+var generators = require('yeoman-generator')
+var _ = require('lodash')
```
Comment (sashanka) [inline generators/app/index.js:2]: Use meaningful variable names. Let minification do the magic for you automatically.

File: generators/app/index.js [hunk 3]
```diff
@@ -1,114 +1,109 @@
-var generators = require('yeoman-generator');
-var _ = require('lodash');
+var generators = require('yeoman-generator')
```
Comment (sashanka) [inline generators/app/index.js]: Call it yeomanGenerator.

File: generators/app/index.js [hunk 4]
```diff
@@ -1,114 +1,109 @@
...
-
-    this.spawnCommand('ls', ['-la']);
-
+    this.option('coffee') // This method adds support for a `--coffee` flag
   },
 
-  _method: function() {
-    this.log('won\'t be called automatically');
-    this.log('Cause method name is prefixed by an underscore (e.g. _method).');
+  _method: function () {
```
Comment (sashanka) [inline generators/app/index.js]: Are you overriding something here? 'method' is a terrible terrible name. Please name your methods meaningfully.

File: generators/app/index.js [hunk 5]
```diff
@@ -1,114 +1,109 @@
...
   },
 
-  askForAttribute: function() {
-    var done = this.async();
-    this.prompt( {
-      type    : 'confirm',
-      name    : 'newAttribute',
-      message : 'Do you wanna add new attribute',
-      default : true
+  askForAttribute: function () {
```
Comment (sashanka) [inline generators/app/index.js]: promptForUserInput. 

And mark this methods as @private or @public accordingly.JSDOC

File: generators/app/index.js [hunk 6]
```diff
@@ -1,114 +1,109 @@
...
-      }.bind(this) );
-
-    }.bind( this ) );
+      this._promptObjectAttribute(answers, function () {
+        this._addAttribute(answers, arguments, done)
+      }.bind(this))
+    }.bind(this))
   },
 
   _writing: function () {
```
Comment (sashanka) [inline generators/app/index.js:257]: better method name please

**Comment by sashanka** [comment]
Please convet to Es6 wherever possible for example:
https://github.com/lukehoban/es6features#template-strings

File: generators/app/index.js (hunk 1 above)
Comment (igorlima) [inline generators/app/index.js]: Details added: https://github.com/igorlima/generator-js-type/commit/26a3aa8fc4cf6830e572bd5c6dfbd688a09d31b6

@sashanka let me know if it's good enough. thanks

File: generators/app/index.js (hunk 2 above)
Comment (igorlima) [inline generators/app/index.js:2]: Done: https://github.com/igorlima/generator-js-type/commit/189384197d0cfaf7f9e9c6d7e3c9d8e3054c8ac6

Thanks

File: generators/app/index.js (hunk 3 above)
Comment (igorlima) [inline generators/app/index.js]: Done: https://github.com/igorlima/generator-js-type/commit/189384197d0cfaf7f9e9c6d7e3c9d8e3054c8ac6

Thanks

File: generators/app/index.js (hunk 4 above)
Comment (igorlima) [inline generators/app/index.js]: Done: https://github.com/igorlima/generator-js-type/commit/cb40f348c111b18bdc5db5680afa9290c036e7f8

Thanks

File: generators/app/index.js (hunk 5 above)
Comment (igorlima) [inline generators/app/index.js]: Done: https://github.com/igorlima/generator-js-type/commit/06b9a8969af06b90c68aec804a51e8a2daa4509b

Thanks

File: generators/app/index.js (hunk 6 above)
Comment (igorlima) [inline generators/app/index.js:257]: Done: https://github.com/igorlima/generator-js-type/commit/80e1d42987593bd9dca07620cf0a5473ce38c4e0

Thanks

**Comment by igorlima** [comment]
converted to Es6 wherever possible:
- Template Strings: https://github.com/igorlima/generator-js-type/commit/96ccddf4bcacee6db7cdb18c8899bf82240a838b
- Arrows: https://github.com/igorlima/generator-js-type/commit/d12f19cb5a3f2874f5b847c47f752f01d7acff9a

**Pull request merged**